stablehlo/dialect/Base.h

```diff
@@ -110,11 +110,26 @@ LogicalResult inferMostSpecificTypeComponents(
     std::optional<Location> location, TypeRange inputTypes,
     SmallVectorImpl<ShapedTypeComponents> &inferredReturnShapes);
 
+template <typename T>
+bool matchElementType(Type elementType) = delete;
+
+template <>
+inline bool matchElementType<int64_t>(Type elementType) {
+  // FIXME: Discuss if this should be signless 64?
+  return elementType.isSignlessInteger();
+}
+
+template <>
+inline bool matchElementType<APInt>(Type elementType) {
+  return elementType.isa<IntegerType>();
+}
+
 // Matches a constant tensor with integer values into a 1-dimensional vector.
 template <typename T>
 LogicalResult matchInts(Value value, SmallVector<T> &result) {
   DenseIntElementsAttr attr;
   if (!matchPattern(value, m_Constant(&attr))) return failure();
+  if (!matchElementType<T>(attr.getElementType())) return failure();
```

Comment:
"Converting to uint asserted when passing a SmallVector<int64_t> to matchInts". The crash was actually caused by something else - namely by `DenseIntElementsAttr::get` encountering a mismatch between the provided type (uint-based) and the provided values (int64_t-based).

I think it would be better to fix the crash at the source - see https://github.com/openxla/stablehlo/pull/1452/commits/ddf7e6895cdb725e1399c71c0f7308d1cda25f8a for details - and keep matchInts unchanged.